keps/sig-instrumentation/2831-kubelet-tracing/README.md

```diff
@@ -401,28 +408,30 @@ previous answers based on experience in the field._
 
   The tracing client library has a small, in-memory cache for outgoing spans.
 
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+  No.
+
 ### Troubleshooting
 
 The Troubleshooting section currently serves the `Playbook` role. We may consider
 splitting it into a dedicated `Playbook` document (potentially with some monitoring
 details). For now, we leave it here.
 
-_This section must be completed when targeting beta graduation to a release._
-
 ###### How does this feature react if the API server and/or etcd is unavailable?
 
   No reaction specific to this feature if API server and/or etcd is unavailable.
 
 ###### What are other known failure modes?
 
-  - [The controller is misconfigured and cannot talk to the collector or the collector cannot send traces to the backend]
+  - [The kubelet is misconfigured and cannot talk to the collector or the kubelet cannot send traces to the backend]
     - Detection: How can it be detected via metrics? Stated another way:
       how can an operator troubleshoot without logging into a master or worker node?
       **kubelet logs, component logs, collector logs**
```

Comment:
Should the kubelet fail to start in this scenario?
Should we ad a node.Status.Condition?

IMVHO non-critical node misconfiguration, like this one seems to be, should be recorded in the node status conditions or any equivalent mechanism to signal the cluster the node can still run, but it is in degraded state wrt the expected configuration.

But it's important to point out that if we want to go this direction, we have a bunch of similar cases already pertaining to GA KEPs, so it would probably better to pursue this effort in a followup, related but distinct, KEP.
Just my 2c.

Reply:
This could be a kubelet misconfiguration, but could also be an issue with the collector.  E.g. it could be temporarily unavailable, or undergoing an upgrade.  I don't think it is a good idea to add additional status updates.  We will have self-observability metrics eventually, which I think is a more appropriate approach.

Reply:
Multiple points of failure might warrant for an indication in the status subresource, so it's easier to narrow down. But I believe that unlike status, the information we expose, and the way in which it is exposed, is more structured and machine-ingestible in case of O11y signals.

There's additionally, as aforementioned, patterns that don't adhere to this trend, so it'd make sense to stick to self-O11y signals to be safe and follow convention.